ament_cmake_auto/cmake/ament_auto_find_build_dependencies.cmake

```diff
@@ -27,18 +32,35 @@
 # @public
 #
 macro(ament_auto_find_build_dependencies)
-  if(ARGN)
+  cmake_parse_arguments(ARG "" "" "REQUIRED_PACKAGES" ${ARGN})
```

Comment:
Please post links to existing CMake code which does that and I am happy to update them. Note that functions (in contrast to macros) have their own scope and are therefore not affected by this.